drivers/display/display_ili9806e_dsi.c

```diff
@@ -0,0 +1,434 @@
+{
+	return ili9806e_write_reg(dev, reg, &value, 1);
+}
+
+static int ili9806e_write_sequence(const struct device *dev, const struct ili9806e_init_cmd *cmd,
+				   uint8_t nr_cmds)
+{
+	int ret = 0;
+
+	/* Loop through all commands as long as writes are successful */
+	for (int i = 0; i < nr_cmds && ret == 0; i++) {
+		ret = ili9806e_write_reg(dev, cmd->reg, cmd->cmd, cmd->cmd_len);
+		if (ret < 0) {
+			LOG_ERR("Failed writing sequence: 0x%x result: (%d)", cmd->reg, ret);
+			return ret;
+		}
+		cmd++;
+	}
+
+	return ret;
+}
+
+static int ili9806e_config(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	ret = ili9806e_write_sequence(dev, init_cmds, ARRAY_SIZE(init_cmds));
+	if (ret < 0) {
+		return ret;
+	}
+	/* Add a delay, otherwise MADCTL not taken */
+	k_msleep(120);
+
+	/* Exit sleep mode */
+	ret = ili9806e_write_reg(dev, MIPI_DCS_EXIT_SLEEP_MODE, NULL, 0);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Wait for sleep out exit */
+	k_msleep(5);
+
+	/* Set color mode */
+	ret = ili9806e_write_reg_val(dev, MIPI_DCS_SET_PIXEL_FORMAT,
+				     cfg->pixel_format == PIXEL_FORMAT_RGB_565
+					     ? ILITEK_ILI9806E_COLMOD_RGB565
+					     : ILITEK_ILI9806E_COLMOD_RGB888);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* Turn on display */
+	ret = ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+	return ret;
+}
+
+static int ili9806e_blanking_on(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 0);
+		if (ret) {
+			LOG_ERR("Disable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_OFF, NULL, 0);
+}
+
+static int ili9806e_blanking_off(const struct device *dev)
+{
+	const struct ili9806e_config *cfg = dev->config;
+	int ret;
+
+	if (cfg->backlight.port != NULL) {
+		ret = gpio_pin_set_dt(&cfg->backlight, 1);
+		if (ret) {
+			LOG_ERR("Enable backlight failed! (%d)", ret);
+			return ret;
+		}
+	}
+
+	return ili9806e_write_reg(dev, MIPI_DCS_SET_DISPLAY_ON, NULL, 0);
+}
+
+static void ili9806e_get_capabilities(const struct device *dev,
+				      struct display_capabilities *capabilities)
+{
+	const struct ili9806e_config *cfg = dev->config;
+
+	memset(capabilities, 0, sizeof(struct display_capabilities));
+	capabilities->x_resolution = cfg->width;
+	capabilities->y_resolution = cfg->height;
+	capabilities->supported_pixel_formats = cfg->pixel_format;
+	capabilities->current_pixel_format = cfg->pixel_format;
+}
+
+static int ili9806e_pixel_format(const struct device *dev,
+				 const enum display_pixel_format pixel_format)
+{
+	const struct ili9806e_config *config = dev->config;
+
+	if (pixel_format == config->pixel_format) {
+		return 0;
+	}
+	LOG_WRN("Pixel format change not implemented");
+
+	return -ENOTSUP;
+}
+
+static const struct display_driver_api ili9806e_api = {
```

Comment:
Zephyr has introduced iterable sections for device APIs, and drivers should use the wrapper macros now.

```suggestion
static DEVICE_API(display, ili9806e_api) = {
```

Reply:
I fixed it